### PR TITLE
Only enable inbound protocol sniffing on ports that need it

### PR DESCRIPTION
**Pull request by howardjohn**
Design doc: https://docs.google.com/document/d/1Jt1IJjNxec4FtIHzwmoCMjRwOX8RNWGoVxawB9_yXro

Mostly done, just need to add some comments throughout the code

### CODE REVIEW

**Comment by howardjohn** [comment]
cc @mandarjog @duderino @bianpengyuan @lambdai

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -108,9 +114,15 @@ func isBindtoPort(l *listener.Listener) bool {
 	return bp.Value
 }
 
+// enabledInspector captures if a given listener needs listener filter inspectors added
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:117]: nit: enabledInspector captures if for a given listener, listener filter inspectors are added

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 2]
```diff
@@ -119,12 +131,29 @@ func reduceInboundListenerToFilterChains(listeners []*listener.Listener) ([]*lis
...
 		for _, c := range l.FilterChains {
-			newChain, needTLSLocal := amendFilterChainMatchFromInboundListener(golangproto.Clone(c).(*listener.FilterChain), l, needTLS)
-			chains = append(chains, newChain)
-			needTLS = needTLS || needTLSLocal
+			chain := golangproto.Clone(c).(*listener.FilterChain)
+			inspectors := amendFilterChainMatchFromInboundListener(chain, l)
+			chains = append(chains, chain)
+			// Aggregate the inspector options. If any listener on the port needs inspector, we should add it
+			// Generally there is 1 listener per port anyways.
+			if l.Address.GetSocketAddress().GetPortValue() > 0 {
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:139]: nit: capture port once and use

**Review by ramaraochavali — COMMENTED** [review]
Couple nits.

**Comment by howardjohn** [comment]
@ramaraochavali I addressed comments

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**